### ISSUE TITLE
FR: [ssh] Use unqualified Windows username

### ISSUE THREAD

**Issue by linusbrogan** [tailscale/tailscale#7362]
### What are you trying to do?

When I run `tailscale.exe ssh [hostname]` on Windows, Tailscale attempts to log on using my qualified [down-level logon name](https://learn.microsoft.com/en-us/windows/win32/secauthn/user-name-formats#down-level-logon-name) (`DOMAIN\username`) instead of my unqualified username (just the `username` part, no `DOMAIN\`) as I expect.
The SSH command fails because my account on the remote host does not include the `DOMAIN\` part in the username.

Example from my Windows machine BOND:
```
C:\>"C:\Program Files (x86)\Tailscale IPN\tailscale.exe" ssh euro
failed to look up BOND\linus
BOND\\linus@euro.tapir-toad.ts.net.: Permission denied (tailscale).
```

### How should we solve this?

Tailscale gets the username here: https://github.com/tailscale/tailscale/blob/abc874b04e85619afeed5f187f6b6c150f4eefbe/cmd/tailscale/cli/ssh.go#L61
On Windows, `lu.Username` is in the down-level logon name format. My first thought is to cut out everything before the backslash when setting `username`. Would that be a problem for anyone with a backslash in their username?

### What is the impact of not solving this?

I either specify the username explicitly (`tailscale.exe ssh [username]@[host]`) or invoke ssh directly (`ssh.exe [host]`), which uses the short username as I expect. Super easy!

### Anything else?

- OS: Windows 11 Home, Version 22H2 (OS Build 22621.1265)
- Tailscale version 1.36.2

<3

**Comment by DentonGentry** [comment]
Just to check: is there a reason to use `tailscale.exe ssh` versus any other SSH client which will populate the username you want it to use?

The crucial bits of Tailscale SSH are implemented within the destination server, not the SSH client. You must:

1. Have run `tailscale up --ssh` on the destination device.
1. Have an ACL policy to allow SSH access
1. You can use any SSH client, but you must ssh to the 100.x.y.z Tailscale IP address or a DNS name which resolves to the Tailscale IP address.

**Comment by linusbrogan** [comment]
I can point you to [`ssh.go`](https://github.com/tailscale/tailscale/blob/abc874b04e85619afeed5f187f6b6c150f4eefbe/cmd/tailscale/cli/ssh.go#L38) for the distinctions between `tailscale ssh` and any other SSH client, but none of those are crucial to me. Automatically checking the host key is nice for the first connection though.
I really don't have any reason to prefer `tailscale ssh` over plain `ssh` other than habit.

**Comment by DentonGentry** [comment]
We can probably check if the platform is Windows and adjust the username before initiating the SSH connection.

**Comment by sammy123k** [comment]
Hello. I just wanted to reach out and check in on this issue. This is an issue that also affects my environment (mixed Windows/Linux setting), and I see that there is a commit here that seems to address the problem. Is there anything holding us back from opening a PR for the branch that @linusbrogan kindly made? I've been a user of Tailscale, but not yet a contributor, so don't want to act like I know too much on the broader implications or the general process here, but on the surface this seems like a relatively benign change to nudge forward?

**Comment by zofrex** [comment]
As [I wrote](https://github.com/tailscale/tailscale/pull/19358#issuecomment-4679289316) in the PR, I don't think #19358 does address this.

Given if we don't pass a username to OpenSSH it still adds the `domain\` to the user, and that we no longer pass a user to SSH at all, I do wonder if this is maybe out of scope, but will leave this open for now for discussion.

**Comment by linusbrogan** [comment]
On recent software versions, this behaves as I expected. Behavior may depend on OpenSSH version. Closing as fixed (or out of scope after #19358).
